logging/log.go

```diff
@@ -391,7 +391,7 @@ func (l logger) GetTelemetrySession() string {
 }
 
 func (l logger) GetTelemetryHostName() string {
-	if !l.GetTelemetryEnabled() {
+	if l.loggerState.telemetry == nil {
```

Comment:
isn't this change effectively rolling back to the previous implementation ?
If so, keeping  `!l.GetTelemetryEnabled()` would be cleaner.